modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -253,9 +254,17 @@
   [field unit]
   (if (= unit :default)
     field
-    (let [column field]
+    (let [date-trunc? (-> (get-mongo-version)
+                          :semantic-version
+                          (driver.u/semantic-version-gte [5]))
+          column field]
       (letfn [(truncate [unit]
-                (truncate-to-resolution column unit))]
+                (if date-trunc?
+                  {:$dateTrunc {:date column
+                                :unit (name unit)
+                                :timezone (qp.timezone/results-timezone-id)
+                                :startOfWeek (name (setting/get-value-of-type :keyword :start-of-week))}}
```

Comment:
You should do `(public-settings/start-of-week)` instead to get this. `setting/get-value-of-type` should mostly be used as a lower-level function to use when implementing setting `:getter` functions and stuff like that. 

There are a few good reasons you want to use the Setting function instead of `setting/get-value-of-type`:

1. `get-value-of-type` will fail if the namespace that contains the `defsetting` wasn't loaded yet somewhere else (race condition)
2. You can jump to the definition of the Setting easily if you use the getter function, but with `get-value-of-type` it's not always clear where the Setting was defined and you have to grep the codebase for it
3. `get-value-of-type` bypasses the Setting's custom `:getter` function. `start-of-week` does not currently have a custom `:getter` function, but if we were to add one in the future, this would be bypassing it

Reply:
I was sloppy.